### ISSUE TITLE
ICC18 binary mode broken (32 and 64-bit)

### ISSUE THREAD

**Issue by pcordes** [mattgodbolt/compiler-explorer#896]
https://godbolt.org/g/E5JroN

64-bit output:

```
/opt/compiler-explorer/gcc-6.3.0/bin/../lib/gcc/x86_64-linux-gnu/6.3.0/../../../../x86_64-linux-gnu/bin/ld: cannot find -lm
Compiler returned: 1
```

`-m32` output:

```/opt/compiler-explorer/gcc-6.3.0/bin/../lib/gcc/x86_64-linux-gnu/6.3.0/../../../../x86_64-linux-gnu/bin/ld: cannot find -lstdc++
/opt/compiler-explorer/gcc-6.3.0/bin/../lib/gcc/x86_64-linux-gnu/6.3.0/../../../../x86_64-linux-gnu/bin/ld: cannot find -lstdc++
/opt/compiler-explorer/gcc-6.3.0/bin/../lib/gcc/x86_64-linux-gnu/6.3.0/../../../../x86_64-linux-gnu/bin/ld: skipping incompatible /opt/compiler-explorer/gcc-6.3.0/bin/../lib/gcc/x86_64-linux-gnu/6.3.0//libgcc.a when searching for -lgcc
/opt/compiler-explorer/gcc-6.3.0/bin/../lib/gcc/x86_64-linux-gnu/6.3.0/../../../../x86_64-linux-gnu/bin/ld: cannot find -lgcc
/opt/compiler-explorer/gcc-6.3.0/bin/../lib/gcc/x86_64-linux-gnu/6.3.0/../../../../x86_64-linux-gnu/bin/ld: cannot find -lgcc_s
Compiler returned: 1
```

**Comment by RabsRincon** [comment]
Hi @pcordes thanks for the report!
This would be the 5th time I "fix" ICC just to later find out that it is, in fact, still broken somehow. 
I should either leave this fix to someone else more capable, or get some help as to where to look for a fix (Although the fix on itself would be pretty great)
It does not seem like we're the ones calling `ld` with `-lm` so it must be ICC internally. I'm guessing there is a way to control what compatibility we want or something like that.
Any help would be appreciated :)

**Comment by mattgodbolt** [comment]
Curiously this works for the newly-installed icc19. I wonder what might be going on here: I was given a new way to install the intel compilers, so maybe there's something in the packaging of the previous versions that was broken?

**Comment by mattgodbolt** [comment]
Comparing with `-Wl,--verbose` (from some `ld` research I was doing) and the icc18 (https://gcc.godbolt.org/z/TSZhgh) uses a `SEARCH_DIR` of:

```
SEARCH_DIR("/root/staging/x86_64-linux-gnu/lib64"); SEARCH_DIR("/root/staging/lib64"); SEARCH_DIR("/usr/local/lib64"); SEARCH_DIR("/lib64"); SEARCH_DIR("/usr/lib64"); SEARCH_DIR("/root/staging/x86_64-linux-gnu/lib"); SEARCH_DIR("/root/staging/lib"); SEARCH_DIR("/usr/local/lib"); SEARCH_DIR("/lib"); SEARCH_DIR("/usr/lib");
```

So! When I first installed icc16 and icc17 I didn't install them to their final resting place (`/opt/compiler-explorer/intel-...`) - I instead installed them to a staging area in a docker container. Then I moved them out (after packing them up as a tar.xz for safekeeping). Clearly there's some path hard-coding going on there and so I'll need to find a way to fix this. Why it isn't just trivially relocatable I don't know...
